homeassistant/components/toon/sensor.py

```diff
@@ -1,202 +1,91 @@
-from homeassistant.helpers.typing import HomeAssistantType
-
-from . import (
+from homeassistant.core import HomeAssistant
+
+from .const import (
+    ATTR_DEFAULT_ENABLED,
+    ATTR_DEVICE_CLASS,
+    ATTR_ICON,
+    ATTR_MEASUREMENT,
+    ATTR_NAME,
+    ATTR_SECTION,
+    ATTR_UNIT_OF_MEASUREMENT,
+    DOMAIN,
+    SENSOR_ENTITIES,
+)
+from .coordinator import ToonDataUpdateCoordinator
+from .models import (
     ToonBoilerDeviceEntity,
-    ToonData,
     ToonElectricityMeterDeviceEntity,
     ToonEntity,
     ToonGasMeterDeviceEntity,
     ToonSolarDeviceEntity,
 )
-from .const import CURRENCY_EUR, DATA_TOON, DOMAIN, VOLUME_CM3, VOLUME_M3
 
 _LOGGER = logging.getLogger(__name__)
 
 
 async def async_setup_entry(
-    hass: HomeAssistantType, entry: ConfigEntry, async_add_entities
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities
 ) -> None:
     """Set up Toon sensors based on a config entry."""
-    toon = hass.data[DATA_TOON][entry.entry_id]
+    coordinator = hass.data[DOMAIN][entry.entry_id]
 
     sensors = [
-        ToonElectricityMeterDeviceSensor(
-            toon, "power", "value", "Current Power Usage", "mdi:power-plug", POWER_WATT
-        ),
-        ToonElectricityMeterDeviceSensor(
-            toon,
-            "power",
-            "average",
-            "Average Power Usage",
-            "mdi:power-plug",
-            POWER_WATT,
-        ),
-        ToonElectricityMeterDeviceSensor(
-            toon,
-            "power",
-            "daily_value",
-            "Power Usage Today",
-            "mdi:power-plug",
-            ENERGY_KILO_WATT_HOUR,
-        ),
-        ToonElectricityMeterDeviceSensor(
-            toon,
-            "power",
-            "daily_cost",
-            "Power Cost Today",
-            "mdi:power-plug",
-            CURRENCY_EUR,
-        ),
-        ToonElectricityMeterDeviceSensor(
-            toon,
-            "power",
-            "average_daily",
-            "Average Daily Power Usage",
-            "mdi:power-plug",
-            ENERGY_KILO_WATT_HOUR,
-        ),
-        ToonElectricityMeterDeviceSensor(
-            toon,
-            "power",
-            "meter_reading",
-            "Power Meter Feed IN Tariff 1",
-            "mdi:power-plug",
-            ENERGY_KILO_WATT_HOUR,
-        ),
-        ToonElectricityMeterDeviceSensor(
-            toon,
-            "power",
-            "meter_reading_low",
-            "Power Meter Feed IN Tariff 2",
-            "mdi:power-plug",
-            ENERGY_KILO_WATT_HOUR,
-        ),
+        ToonElectricityMeterDeviceSensor(coordinator, key=key)
+        for key in {
+            "power_average_daily",
+            "power_average",
+            "power_daily_cost",
+            "power_daily_value",
+            "power_meter_reading_low",
+            "power_meter_reading",
+            "power_value",
+            "solar_meter_reading_low_produced",
+            "solar_meter_reading_produced",
+        }
```

Comment:
No need to use a set.

```suggestion
        for key in (
            "power_average_daily",
            "power_average",
            "power_daily_cost",
            "power_daily_value",
            "power_meter_reading_low",
            "power_meter_reading",
            "power_value",
            "solar_meter_reading_low_produced",
            "solar_meter_reading_produced",
        )
```